sidecar/internal/service/reclaimspace.go

```diff
@@ -61,12 +66,29 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	var deviceSecret map[string]string
+	var volID string
+	var volAttributes map[string]string
+
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	if pv.Spec.CSI != nil {
```

Comment:
better check

```go
if pv.Spec.CSI == nil {
    return nil, status.Errorf(codes.InvalidArgument, "pv %q is not a CSI volume", pvName)
}

volID = ...
```

Reply:
The check seems to be incorrect. You need to check for `pv.Spec.CSI == nil` as that is the error case.

Reply:
This still needs to be fixed. Once done, I am happy to approve this.